src/openrct2-ui/windows/Footpath.cpp

```diff
@@ -871,11 +959,22 @@ static void window_footpath_place_path_at_point(const ScreenCoordsXY& screenCoor
         slope &= ~RAISE_FOOTPATH_FLAG;
         z += PATH_HEIGHT_STEP;
     }
-    auto selectedType = (_footpathSelectedType << 7) + (gFootpathSelectedId & 0xFF);
 
     // Try and place path
     gGameCommandErrorTitle = STR_CANT_BUILD_FOOTPATH_HERE;
-    auto footpathPlaceAction = FootpathPlaceAction({ info.Loc, z }, slope, selectedType);
+    auto selectedType = gFootpathSelection.GetSelectedSurface();
+    PathConstructFlags constructFlags = 0;
```

Comment:
This same code is used a number of times. Suggest introducing a function for it.